### ISSUE TITLE
Can you add a method, let us can use SDWebImage to get an image  correct file size？

### ISSUE THREAD

**Issue by housenkui** [SDWebImage/SDWebImage#2677]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

https://stackoverflow.com/questions/13363563/how-to-get-uiimage-correct-file-size.
Can you add a method, let us can use SDWebImage to get an image  correct file size？ Thanks very much!!!
I have try my best to do it. but do not work...

**Comment by Insofan** [comment]
This feature is not for SD‘s design purpose, and maybe you could get some idea from [UIImage+MemoryCacheCost.m](https://github.com/SDWebImage/SDWebImage/blob/master/SDWebImage/UIImage%2BMemoryCacheCost.m) . Actually you just need a UIImage category to calculate image size.It isn't very complicated, reference [Stackoverflow link](https://stackoverflow.com/questions/1296707/get-size-of-a-uiimage-bytes-length-not-height-and-width/7500214#7500214)

**Comment by dreampiggy** [comment]
@housenkui I think maybe you misunderstand, what it image (bitmap), and what is the compressed image data.

# Image Bitmap
For example, if you care about the memory usage of image, you should concentrate on the bitmap byte size. Which is just a array of pixles (for 32bit color, each pixel just means a UInt8[4])

The bitmap image memory usage, it's only apply for resolution and color components.

```
Bitmap Bytes = pixelHeight * pixelWidth * bytesPerPixel (for 32bit color, it's 4Bytes)
```

# Compressed Image Data
Because raw image bitmap it's really large, and actually there are always some same pixels, we can use the compression algorithm to produce a data suitable for transfer from Network. It's the compressed image data.

All JPEG/PNG/GIF/WebP/HEIF, are all detailed compressed image data. Which use the different compression algorithm for the raw pixel image bitmap, and contains some extra information about image metainfo.

# SDWebImage and this two type

For SDWebImage, if you're careing about the image bitmap size, you can calculate it from the image bitmap representation `UIImage` or `CGImage`. The demo code:

```objective-c
CGImageRef imageRef;
NSUInteger bytesSize = CGImageGetBytesPerRow(imageRef) * CGImageGetHeight(imageRef);
```

You can even use the free convient method, the `sd_memoryCost` getter in `UIImage+MemoryCacheCost.h`. Because for Animated Images, it contains multiple frames, this also handler the cases.

For SDWebImage, if you care about the compressed image data size, you know, you must have a `NSData` represent of the compressed data.

For image already in disk cache, you can grab the image data using:

```objective-c
NSString *key;
NSData *imageData = [SDImageCache.sharedImageCache diskImageDataForKey:key];
NSUInteger bytesSize = imageData.length
```

If you just download the image data from network or some custom loader, you can grab the `NSData` arg in the completionBlock

**Comment by housenkui** [comment]
@dreampiggy @Insofan  Thanks very much.
